java/src/jmri/InstanceManagerAutoInitialize.java

```diff
@@ -0,0 +1,23 @@
+package jmri;
+
+/**
+ * Provide a hint to the {@link jmri.InstanceManager} that this object needs
+ * have additional initialization performed after it is made available by the
+ * InstanceManager. This allows two classes that have circular dependencies on
+ * being able to get the default instance of each other to be managed
+ * successfully.
+ * <p>
+ * Note: the need to have a class implement this probably is indicative of other
+ * design issues in the implementing class.
+ *
+ * @author Randall Wood Copyright 2017
+ */
+public interface InstanceManagerAutoInitialize {
```

Comment:
From the code, I think is is only when InstanceManager takes responsibility to create it, not when added. OK, got that.

What's an example of the sequence it's trying to fix?  The attached isn't quite it, I think. Maybe it's A asks for B which asks for A?

![instancesequence](https://user-images.githubusercontent.com/2774117/29006433-8d21c060-7aa4-11e7-96c4-570870ec1141.png)
[InstanceSequence.txt](https://github.com/JMRI/JMRI/files/1203373/InstanceSequence.txt)